src/custom/components/AccountDetails/index.tsx

```diff
@@ -38,22 +38,44 @@ import {
   IconWrapper,
   renderTransactions
 } from './AccountDetailsMod'
+import { ConnectedWalletInfo, useWalletInfo } from 'hooks/useWalletInfo'
+import { MouseoverTooltip } from 'components/Tooltip/TooltipMod'
 
 type AbstractConnector = Pick<ReturnType<typeof useActiveWeb3React>, 'connector'>['connector']
 
-export function formatConnectorName(connector?: AbstractConnector) {
+export function formatConnectorName(connector?: AbstractConnector, walletInfo?: ConnectedWalletInfo) {
   const { ethereum } = window
+  const { walletName } = walletInfo || {}
   const isMetaMask = !!(ethereum && ethereum.isMetaMask)
-  const name = Object.keys(SUPPORTED_WALLETS)
-    .filter(
-      k => SUPPORTED_WALLETS[k].connector === connector && (connector !== injected || isMetaMask === (k === 'METAMASK'))
-    )
-    .map(k => SUPPORTED_WALLETS[k].name)[0]
+  const name =
+    walletName ||
+    Object.keys(SUPPORTED_WALLETS)
+      .filter(
+        k =>
+          SUPPORTED_WALLETS[k].connector === connector && (connector !== injected || isMetaMask === (k === 'METAMASK'))
```

Comment:
I know you didn't write this but this line sucks to read lol

Reply:
Agreed 😮‍💨

Will see what I can do

Reply:
Gave it a try on https://github.com/gnosis/gp-swap-ui/pull/633/commits/0d2b9e4135ab8d8ef113a361f671c2326bc0e225

Not very happy with it yet, but I've refactored some of the ugliness away